apps/prairielearn/src/lib/assessment.ts

```diff
@@ -437,3 +437,21 @@ export async function selectAssessmentInstanceLogCursor(
     InstanceLogSchema,
   );
 }
+
+export async function updateAssessmentQuestionStats(assessment_question_id: string): Promise<void> {
+  await sqldb.queryAsync(sql.calculate_stats_for_assessment_question, { assessment_question_id });
+}
+
+export async function updateAssessmentQuestionStatsForAssessment(
+  assessment_id: string,
+): Promise<void> {
+  await sqldb.runInTransactionAsync(async () => {
+    const assessment_questions = await sqldb.queryRows(
+      sql.select_and_lock_assessment_questions,
+      { assessment_id },
+      IdSchema,
+    );
```

Comment:
I believe locking is a change in behavior here, is that right? I'd like input from @mwest1066 on this; I'm a little wary about locking on so many rows for what might be an expensive operation.

Reply:
The original sproc did not select for locking, but if called the sproc on each row, updating them as they were selected. The locking addition is indeed a change in behaviour (I didn't notice that originally), but in effect all assessment questions were updated (and I assume locked as a result) in the same transaction, so I'm not sure how different this would be in terms of final result. I don't mind restoring the query if necessary, I'm just wandering if that's a big difference.

Reply:
If I'm understanding PG locking correctly, I don't think it will make much difference in practice whether we select with a lock or not. We'll lock each `assessment_questions` row anyway when we update its statistics and that lock will be held until the end of the transaction, so I think it's only a difference between:
1. With explicit lock: we lock all assessment questions from the start of the transaction to the end.
2. Without explicit lock: Q1 is locked from the start of the transaction to the end, Q2 is locked from a little bit later until the end of the transaction, etc.

For code simplicity I'm tempted to remove the explicit lock?